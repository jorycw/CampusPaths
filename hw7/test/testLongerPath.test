# Find a path with Dijkstra's algorithm when there are two distinct paths
# in which the path containing more edges has a lower cost

CreateGraph graph
AddNode graph a
AddNode graph b
AddNode graph c
AddNode graph d

AddNode graph a.5

AddEdge graph a a.5 .2
AddEdge graph a.5 b .1
AddEdge graph b d 2.2

AddEdge graph a c 2
AddEdge graph c d 2

FindPath graph a d