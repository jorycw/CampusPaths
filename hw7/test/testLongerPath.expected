# Find a path with Dijkstra's algorithm when there are two distinct paths
# in which the path containing more edges has a lower cost

created graph graph
added node a to graph
added node b to graph
added node c to graph
added node d to graph

added node a.5 to graph

added edge 0.200 from a to a.5 in graph
added edge 0.100 from a.5 to b in graph
added edge 2.200 from b to d in graph

added edge 2.000 from a to c in graph
added edge 2.000 from c to d in graph

path from a to d:
a to a.5 with weight 0.200
a.5 to b with weight 0.100
b to d with weight 2.200
total cost: 2.500